pkg/istio-agent/xds-proxy.go

```diff
@@ -247,6 +257,23 @@ func (p *XdsProxy) close() {
 	}
 }
 
+// isExpectedGRPCError checks a gRPC error code and determines whether it is an expected error when
+// things are operating normally. This is basically capturing when the client disconnects.
+func isExpectedGRPCError(err error) bool {
+	if err == io.EOF {
+		return true
+	}
+
+	s := status.Convert(err)
+	if s.Code() == codes.Canceled || s.Code() == codes.DeadlineExceeded {
+		return true
+	}
+	if s.Code() == codes.Unavailable && (s.Message() == "client disconnected" || s.Message() == "transport is closing ") {
```

Comment:
` transport is closing ` has a wider meaning https://chromium.googlesource.com/external/github.com/grpc/grpc-go/#the-rpc-failed-with-error

Reply:
Yes. It has but among all reasons , I think only #3 is applicable - not others. We will catch others much early except tcp keep alive which we do not set.  Do you see any alternative way of catching this?